### ISSUE TITLE
Proposal: Include lighweight-styling resources in API spec and in the API review process

### ISSUE THREAD

**Issue by Felix-Dev** [microsoft/microsoft-ui-xaml#2916]
# Proposal: Include lighweight-styling resources in API spec and in the API review process

## Summary
Currently, the lighweight styling resources a control offers to developers are not part of the API spec for that control. Lightweight styling resources are part of API surface of the control though and thus should be covered in the API spec and take part in the API review process. Take a look for example at the [TabView API spec](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/master/archive/Microsoft.UI.Xaml%202.2/tab/tabcontrol.md) or the [NumberBox spec](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/savoyschuler/numberbox/active/NumberBox/NumberBox.md). None of the specs include any lightweight styling resource!

This has already come back to bite us in quite a few cases, for example in the [TabView case](https://github.com/microsoft/microsoft-ui-xaml/issues/2742). The NumberBox control also currently ships with a [suboptimal lightweight-styling experience](https://github.com/microsoft/microsoft-ui-xaml/issues/2844). Lightweight styling of other controls like the NavigationView also suffers fromincorrect and inconsistent resource naming. 

Due to [limitations in the current XAML resource lookup system](https://github.com/microsoft/microsoft-ui-xaml/issues/2913) updating the used (theme) resources *after* an initial version of the control has already been shipped to customers is proving to be problematic and forces us to ship WinUI with a suboptimal lightweight-styling experience.

## Rationale
Integrating lightweight-styling resources as part of the API spec and thus as part of the API review process will greatly reduce the situation where we ship a control with *inconsistent, incorrect and missing* (theme) resources and thus provide a high-quality lightweight-styling experience from the get-go and expected by developers.

The lightweight-styling resources so far have lacked significantly in receiving the great care which other public APIs are enjoying as part of the API review. As lightweight-styling plays an important rule in WinUI we should start handling them with the same level of care as the rest of the public control API surface as soon as possible.

## Important Notes
We should start applying this to controls currently being "spec'ed" or in the process of being developed, like the [Pager]((https://github.com/Microsoft/microsoft-ui-xaml-specs/blob/user/savoyschuler/pagercontrol/active/PagerControl/PagerControl.md) control or the [InfoBar](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/gabilka/infobar/active/InformationBar/InformationBar.md) control!

**Comment by ranjeshj** [comment]
I agree. lightweight styling resources are akin to public API and changes can end up as breaking changes. Unfortunately this is not exposed through IDL and is easy to change in the repo. It would be great if there is typed APIs for the resources to make this more obvious.  @MikeHillberg for thoughts on the spec process.
